packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -53,43 +47,71 @@ export const styles = {
   focusVisible: {},
 };
 
+/**
+ * https://github.com/facebook/react/issues/14099#issuecomment-440013892
+ *
+ * @param {function} fn
+ */
+function useEventCallback(fn) {
+  const ref = React.useRef(fn);
+  const useEnhancedEffect = typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;
```

Comment:
You can move it outside of the scope of the function.

Reply:
I'm wondering if only using `React.useEffect` would be enough.

Reply:
Probably not. I'm pretty sure user event will have a higher priority than effects which means we can potentially call stale callbacks. Why not leave the original source as is? It looks like this is officially endorsed from the issue thread.

Reply:
Moved it outside of the scope. But it needs to happen during commit.